library/src/main/java/com/novoda/downloadmanager/DownloadFile.java

```diff
@@ -60,7 +62,16 @@ void download(Callback callback) {
 
         fileSize.setCurrentSize(filePersistence.getCurrentSize(filePath));
 
-        persistSync();
+        if (downloadFileStatus.isMarkedAsDeleted()) {
+            return;
+        }
+
+        Log.v("persist file " + downloadFileId.rawId() + ", with status: " + downloadFileStatus.status());
+        boolean persist = persist();
+        if (!persist) {
```

Comment:
Can we inline the variable? Looks like the only place it is used.

Reply:
yes, I didn't do it in the first place because of the negation, will do